src/gc/gc.cpp

```diff
@@ -5332,18 +5332,15 @@ void gc_heap::gc_thread_function ()
 
             gc_heap::internal_gc_done = true;
 
-            set_gc_done();
-        }
-        else
-        {
-            int spin_count = 32 * (gc_heap::n_heaps - 1);
-
-            // wait until RestartEE has progressed to a stage where we can restart user threads
-            while (!gc_heap::internal_gc_done && !GCHeap::SafeToRestartManagedThreads())
+#ifdef MULTIPLE_HEAPS
+            for (int i = 0; i < gc_heap::n_heaps; i++)
             {
-                spin_and_switch (spin_count, (gc_heap::internal_gc_done || GCHeap::SafeToRestartManagedThreads()));
+                gc_heap::g_heaps[i]->set_gc_done();
```

Comment:
Heap 0 marks all heaps done to eliminate need to spin on RestartEE or wake if we are not proceeding.